modules/javafx.base/src/test/java/test/javafx/beans/property/ObjectPropertyLeakTest.java

```diff
@@ -0,0 +1,278 @@
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleFloatProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleLongProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class ObjectPropertyLeakTest {
+
+    private static final int OBJ_COUNT = 10;
+
+    private final ArrayList<Property<?>> origList = new ArrayList<>();
+    private final ArrayList<Property<?>> wrappedList = new ArrayList<>();
+
+    private final ArrayList<WeakReference<Property<?>>> origRefs = new ArrayList<>();
+    private final ArrayList<WeakReference<Property<?>>> wrappedRefs = new ArrayList<>();
+
+    private void commonLeakTest(int origExpected, int wrappedExpected)
+            throws Exception {
+
+        for (int i = 0; i < 5; i++) {
+            System.gc();
+            System.runFinalization();
+            Thread.sleep(50);
+        }
+
+        int origCount = 0;
+        for (var ref : origRefs) {
+            if (ref.get() != null) origCount++;
+        }
+        final String origMsg = origExpected > 0
+                ? "Original properties should NOT be GCed"
+                : "Original properties should be GCed";
+        assertEquals(origMsg, origExpected, origCount);
+
+        int wrappedCount = 0;
+        for (var ref : wrappedRefs) {
+            if (ref.get() != null) wrappedCount++;
+        }
+        final String wrappedMsg = wrappedExpected > 0
+                ? "Wrapped properties should NOT be GCed"
+                : "Wrapped properties should be GCed";
+        assertEquals(wrappedMsg, wrappedExpected, wrappedCount);
+    }
+
+    private void commonLeakTest() throws Exception {
+        // Verify that we hold references to both original and wrapped objects
+        commonLeakTest(OBJ_COUNT, OBJ_COUNT);
+
+        // Clear references to wrapped property objects and recheck
+        wrappedList.clear();
+        commonLeakTest(OBJ_COUNT, 0);
+
+        // Clear references to original property objects and recheck
+        origList.clear();
+        commonLeakTest(0, 0);
+    }
+
+    @Test
+    public void testBooleanPropertyAsObjectLeak() throws Exception {
+        for (int i = 0; i < OBJ_COUNT; i++) {
+            // Create original and wrapped property objects
+            final BooleanProperty origProp = new SimpleBooleanProperty(true);
+            final ObjectProperty<Boolean> wrappedProp = origProp.asObject();
+
+            // Save reference to original and wrapped objects
+            origList.add(origProp);
+            wrappedList.add(wrappedProp);
+
+            // Save weak references for GC detection
+            origRefs.add(new WeakReference<>(origProp));
+            wrappedRefs.add(new WeakReference<>(wrappedProp));
```

Comment:
I suggest extracting this section into its own method since it's being used in all other tests.

Reply:
Good idea. I'll do that.